workitem/undoable_wi_repo.go

```diff
@@ -65,6 +65,30 @@ func (r *UndoableWorkItemRepository) Save(ctx context.Context, wi app.WorkItem)
 	return res, errs.WithStack(err)
 }
 
+// Reorder implements application.WorkItemRepository
+func (r *UndoableWorkItemRepository) Reorder(ctx context.Context, before string, wi app.WorkItem) (*app.WorkItem, error) {
+	id, err := strconv.ParseUint(wi.ID, 10, 64)
+	if err != nil {
+		// treating this as a not found error: the fact that we're using number internal is implementation detail
+		return nil, errors.NewNotFoundError("work item", wi.ID)
+	}
+
+	old := WorkItem{}
+	db := r.wrapped.db.First(&old, id)
+	if db.Error != nil {
```

Comment:
Agreed. The Internal Error should be checked before NotFounf error. Thanks for pointing it out. i fixed it here -> d0dbec6